src/dstack/_internal/core/backends/vultr/compute.py

```diff
@@ -0,0 +1,161 @@
+import json
+import time
+from typing import List, Optional
+
+import requests
+
+from dstack._internal.core.backends.base import Compute
+from dstack._internal.core.backends.base.compute import (
+    get_instance_name,
+    get_shim_commands,
+)
+from dstack._internal.core.backends.base.offers import get_catalog_offers
+from dstack._internal.core.backends.vultr.api_client import VultrApiClient
+from dstack._internal.core.backends.vultr.config import VultrConfig
+from dstack._internal.core.errors import BackendError, ProvisioningError
+from dstack._internal.core.models.backends.base import BackendType
+from dstack._internal.core.models.instances import (
+    InstanceAvailability,
+    InstanceConfiguration,
+    InstanceOfferWithAvailability,
+    SSHKey,
+)
+from dstack._internal.core.models.runs import Job, JobProvisioningData, Requirements, Run
+from dstack._internal.core.models.volumes import Volume
+from dstack._internal.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class VultrCompute(Compute):
+    def __init__(self, config: VultrConfig):
+        self.config = config
+        self.api_client = VultrApiClient(config.creds.api_key)
+
+    def get_offers(
+        self, requirements: Optional[Requirements] = None
+    ) -> List[InstanceOfferWithAvailability]:
+        offers = get_catalog_offers(
+            backend=BackendType.VULTR,
+            requirements=requirements,
+        )
```

Comment:
Make sure to enforce `regions` if they are specified in the backend configuration.
```suggestion

        offers = get_catalog_offers(
            backend=BackendType.VULTR,
            locations=self.config.regions or None,
            requirements=requirements,
        )
```